Orange/data/sql/backend/mssql.py

```diff
@@ -117,17 +117,26 @@ def _guess_variable(self, field_name, field_metadata, inspect_table):
     EST_ROWS_RE = re.compile(r'StatementEstRows="(\d+)"')
 
     def count_approx(self, query):
-        try:
-            with self.connection.cursor() as cur:
+        with self.connection.cursor() as cur:
+            try:
                 cur.execute("SET SHOWPLAN_XML ON")
                 try:
                     cur.execute(query)
                     result = cur.fetchone()
                     return int(self.EST_ROWS_RE.search(result[0]).group(1))
+                except AttributeError:
+                # This is to catch a bug from SQL Server 2012
```

Comment:
If I understand correctly, the query plan in SQL Server 2012 produces an XML, which does not match the EST_ROWS_RE regexp as it represents the number of rows as float? If this is the case, why not update the regular expression to match both?

Instead of
```
StatementEstRows="(\d+)"
```
which expects an integer between the parentheses, we could use something like this
```
r'StatementEstRows="(\d+)(\.\d*)?"'
```
the second regular expressions allow the number to end with a decimal point and some numbers, but only parses the whole part

Reply:
Not exacty, after searching it seem that our SQL server statistics are not maintained often enough and this results in bad values being returned in form of a float thus the AttributeError. I'm not sure if a well maintained server would return a float: I have a second database where statistics are maintained and I never received a float from the Plan query.
It would require more samples to figure out what to do. This code works in all situations, the drawback is the execution time if a float is returned. Maybe there should be a warning issued like those deprecated warnings found in some libraries saying that "statistics are out of date, SELECT count(*) used." but I do not know how to throw such a message in Orange.

Reply:
I modified the code comments to explain the reason of the AttributeError. If someone ever comes back with a float with the right estimated result, the code will have to be updated as you suggested and if possible with a sanity check on the statistics (so un-maintained statistics would be caught).